packages/esm-active-visits-app/src/active-visits-widget/active-visits.component.tsx

```diff
@@ -244,43 +244,47 @@ const ActiveVisitsTable = () => {
                   </TableRow>
                 </TableHead>
                 <TableBody>
-                  {rows.map((row, index) => (
-                    <React.Fragment key={index}>
-                      <TableExpandRow
-                        {...getRowProps({ row })}
-                        data-testid={`activeVisitRow${activeVisits?.[index]?.patientUuid}`}>
-                        {row.cells.map((cell) => (
-                          <TableCell key={cell.id} data-testid={cell.id}>
-                            {cell.info.header === 'name' ? (
-                              <PatientNameLink
-                                from={fromPage}
-                                to={`\${openmrsSpaBase}/patient/${activeVisits?.[index]?.patientUuid}/chart/`}>
-                                {cell.value}
-                              </PatientNameLink>
-                            ) : (
-                              cell.value
-                            )}
-                          </TableCell>
-                        ))}
-                      </TableExpandRow>
-                      {row.isExpanded ? (
-                        <TableRow className={styles.expandedActiveVisitRow}>
-                          <th colSpan={headers.length + 2}>
-                            <ExtensionSlot
-                              className={styles.visitSummaryContainer}
-                              name="visit-summary-slot"
-                              state={{
-                                visitUuid: activeVisits[index]?.visitUuid,
-                                patientUuid: activeVisits[index]?.patientUuid,
-                              }}
-                            />
-                          </th>
-                        </TableRow>
-                      ) : (
-                        <TableExpandedRow className={styles.hiddenRow} colSpan={headers.length + 2} />
-                      )}
-                    </React.Fragment>
-                  ))}
+                  {rows.map((row, index) => {
+                    const visit = activeVisits.find((visit) => visit.id === row.id);
```

Comment:
This bit of code finds the visit object whose data is represented in the current row (based on the matching `visit id`). Later on, the `patientUuid` gets obtained from this visit data. This is more robust than using array indices to try and determine which visit we're currently looking at.